iosdevapp/ViewModel.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  ModelView.swift
+//  iosdevapp
+//
+//  Created by Leonit Ejupi on 07.02.23.
+//
+
+import Foundation
+
+final class ViewModel: ObservableObject {
+    private let urlSession: URLSession
+    @Published var imageURL: URL?
+    @Published var isLoading = false
+    
+    init(urlSession: URLSession = URLSession.shared) {
+        self.urlSession = urlSession
+    }
+    
+    func generateImage(withText text: String) async {
+        guard let url = URL(string: "https://api.openai.com/v1/images/generations") else {
+            return
+        }
+        var urlRequest = URLRequest(url: url)
+        urlRequest.httpMethod = "POST"
+        urlRequest.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        urlRequest.addValue("Bearer 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", forHTTPHeaderField: "Authorization")
+        
+        let dictionary: [String: Any] = [
+            "prompt": text,
+            "n": 1,
+            "size": "1024x1024"
+        ]
+         
+        urlRequest.httpBody = try! JSONSerialization.data(withJSONObject: dictionary)
```

Comment:
You could use `JSONEncoder` instead of `JSONSerialization` like you used `JSONDecoder` already below. `JSONSerialization` is an older API from objective-C times.